reana-ui/src/pages/signin/SignWrapper.js

```diff
@@ -0,0 +1,43 @@
+/*
+  -*- coding: utf-8 -*-
+
+  This file is part of REANA.
+  Copyright (C) 2020 CERN.
+
+  REANA is free software; you can redistribute it and/or modify it
+  under the terms of the MIT License; see LICENSE file for more details.
+*/
+
+import React from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { useHistory, useLocation } from "react-router-dom";
+
+import { getConfig } from "../../selectors";
+import SignContainer from "./components/SignContainer";
+
+export default function SignWrapper(WrappedComponent) {
+  return function SignWrapperHoc() {
```

Comment:
Opinion: I'm not a big fan of HOC, although I understand that in this particular case can make sense as you're not injecting new props to the component, and I think that the wrapped components can also work standalone. More about this [here](https://medium.com/javascript-scene/do-react-hooks-replace-higher-order-components-hocs-7ae4a08b7b58). 

Have you considered replicating the same with [custom hooks](https://reactjs.org/docs/hooks-custom.html)?
I think that something like `useSign()` that returns `handleSubmit` function can work similarly.

This is not required at all, only in case you wanted to give it a try. I'd like to see some custom hooks in our app at some point.

Reply:
Thanks a lot for this suggestion! Indeed it looks much cleaner, no need to create unnecessary layers of components in this case. Custom hooks didn't came into my mind somehow. I've submitted my changes in additional commit so that it would be easier for you to see the changes. Later I can squash it

Reply:
Looks amazing, so happy to see our first custom hook! 😇
I wonder if it would make sense to have a dedicated file/folder called `hooks` to store them? WDYT?
I see them as helpers/utils that can be reused in different components so it might make sense. Not sure what's the common practice in this case.

Reply:
Totally agree, I had a hard time figuring out where to put it. I moved it to the top level of the project file called `hooks` for now. Later once we have lots of them we can refactor it easily into folder with files.
Actually this hook seems very specific for Sign in/up components, not sure if we will be able to reuse it somehow, but still it's quite nice to have all the hooks in one place!